2015/proposals/agarwal-rajat.md

```diff
@@ -0,0 +1,92 @@
+# Title
+
+Add taxonomic name resolution to the EcoData Retriever to facilitate data science approaches to ecology
+
+## Abstract
+
+The EcoData Retriever is a Python based tool for automatically downloading, cleaning up, and restructuring ecological data. It does the data munging so that the users can focus on using the data.
+As in ecological and evolutionary data, the names of species are constantly being redefined - it is difficult to combine datasets. By automating reconciliation of different species names as part of the process of accessing the data in the first place it will become much easier to combine diverse datasets and in new and interesting ways.
+
+## Technical Details
+
+The project would consists of the following components:
+
+1. Querying the taxonomy name resolution service:
+APIs of the name resolution services have to be used to fetch the possible names using Python. The name of the species would be the input to the APIs and their possible names would be returned. This will be a separate module for the purpose of reusability.      
+
+2. Decision of the best name
+From the possible names returned from different services, an algorithm to determine the best name has to be developed. This would include giving different weights to parameters like reliability of source, consistency with other services and other parameters which need to be thought of. 
+
+3. Updating the database
+Once the decision about the best name is made, this component will update the data model and the user interface with the best name and its location.
+
+4. Building control flow
+This component would allow the user to choose whether to run name resolution or not. Also, name resolution for the species should not run if the species name is not present in the data model. This will act as a bridge between the UI-Data Model module and the Name Resolution module.
+
+
+## Schedule of Deliverables
+
+Before the coding period begins, I intend to do the following:
+1. Get more familiar with the existing code base.
+2. Finding the best name resolution services which are to be used.
+3. Making the design decisions of various modules with the mentors.
+
+### May 25th -  June 7th
+
+Getting the list of names, by interfacing with the different APIs, that are to be given as input to the best name deciding algorithm.  
+
+### June 8th - June 21st
+
+Developing the best name deciding algorithm. 
+
+### June 22nd - July 5th
+
+Updating the data models and the user interfaces with the best name.
```

Comment:
The best name will need to be obtained dynamically when running the retriever (and possibly cached), so I don't think you'd want to try to update the data model or the user interface with the best name, so much as pass the best name to the resulting database when the retriever runs and creates a particular dataset.